And onboarding documentation for workshop (#535)
aicoe-aiops/ocp-ci-analysis: docs/workshop/onboarding.md

# Onboarding

Hello and welcome to the workshop on building a simple AIOps tool on the [Operate First cloud](https://www.operate-first.cloud/) and learning how to move your AI workloads to the cloud by implementing an end-to-end ML tool.

In this workshop, you will learn how to collect data from an open CI/CD data source, use Jupyterlab and its extensions to manage dependencies, and run jupyter notebooks. You will learn how to perform automated, repeatable experiments using Elyra and Kubeflow Pipelines, create and deploy machine learning modes on OpenShift using Seldon and visualize results on a Superset dashboard.

> 

Reviewer: Should we add a section like 
```suggestion
### Access

To get access to the services please add your github username [here](link to some online text editor maybe etherpad?), and you will be provided access.
```

Reviewer: sounds good to me :)

To get started with this tour, let’s first introduce you to the various tools that are available on Operate First environment and will help us in building the AIOps application using Operate First.

1. **Openshift Console -**

The Operate First initiative currently manages two clusters within the [MOC environment](https://massopen.cloud/).

* [Smaug cluster](https://console-openshift-console.apps.smaug.na.operate-first.cloud/) for all user workloads
* [Infra cluster](https://console-openshift-console.apps.moc-infra.massopen.cloud/) for cluster management, housing ACM and ArgoCD

For this workshop, we shall proceed with using the [Smaug cluster](https://console-openshift-console.apps.smaug.na.operate-first.cloud/).

![Cluster Login Page](../assets/images/cluster1.png)

Select login with `operate-first`, and login into the cluster by using your Github credentials.

> ![Cluster Landing Page](../assets/images/cluster2.png)

Author: I have resolved all comments @oindrillac and @suppathak, added a screenshot from a test-user pov.

2. **Ceph S3 storage -**

> Ceph provides a unified storage service with an object, block, and file interface from a single cluster built from commodity hardware components. Operate First hosts a public ceph storage bucket that is accessible by anyone. You can find the credentials and other information for this bucket below:

Reviewer: We need to update these bucket creds to the new bucket. We need to check with Humair if the creds https://console-openshift-console.apps.smaug.na.operate-first.cloud/k8s/ns/aiops-tools-workshop/secrets/s3buckets should be available as markdown here? Might be better to link to the credentials hosted in a text editor for the workshop

Author: Yes, I was waiting on the testing bit before we add those details here. And regarding the creds being available here, this is how operate first gitops documentation has presented these [here](https://www.operate-first.cloud/apps/content/odh/trino/access_public_bucket.html). Other option would be vault or any other text editor we decide on

```
Endpoint:  https://s3-openshift-storage.apps.smaug.na.operate-first.cloud

Bucket Name: opf-datacatalog

Access Key: INDuwCuYntikCsRQioSk

Secret Key: YjLeEDP+Lj30BJTbzgy84b3pzXPpWtlYnRcMYW2/

Total Size: 15G
```

**_Some things to note:_**

* Do NOT store any confidential information on this bucket as it is open to the public
* Always back up all your data regularly, as anyone can go in and wipe it
3. **Jupyterhub -**

JupyterHub is the best way to serve Jupyter notebook for multiple users. It is a multi-user Hub that spawns, manages, and proxies multiple instances of the single-user Jupyter notebook server. Access to operate first JupyterHub is open to the public. All you need is a GitHub account.

> * Our JupyterHub instance on the smaug cluster can be found [here](https://jupyterhub-opf-jupyterhub.apps.smaug.na.operate-first.cloud/).

Reviewer: ```suggestion
* Our JupyterHub instance on the smaug cluster can be found [here](https://jupyterhub-aiops-tools-workshop.apps.smaug.na.operate-first.cloud/).
```

Reviewer: we should have users select the workshop sizing and use this screenshot
<img width="861" alt="Screen Shot 2022-07-18 at 4 50 23 PM" src="https://user-images.githubusercontent.com/32435206/179614984-0a22f25c-2fcf-4d03-a0c8-b08473cf7f76.png">

* Login using the operate-first option, and you will be redirected to the JH spawner UI.

![JH Landing Page](../assets/images/JH_landing_page.png)

> * You can select any notebook image that you want to try on, for this workshop, we will be working with the `ocp-ci-analysis` image.

Reviewer 2: For starters, it might be confusing since `ocp-ci-analysis` name is not in the list in the pic. 

```suggestion
* You can select any notebook image that you want to try on, for this workshop, we will be working with the `Openshift CI Analysis Notebook Image` image.
```

> * Select a suitable size for the server, we would recommend using the `medium` size, and start the server.

Reviewer: We need to check if medium size is what we are recommending. I think the default user profile will point to the custom workshop sizing

Reviewer: The sizing will be called "Workshop"

> * Select a suitable size for the server, we would recommend using the `medium` size, and start the server.

Reviewer: Very good point. We should mention this 👍

> 

Reviewer: Shall we also add a section on Seldon too?

> 4. **Trino -**

Author: Awesome, will update them :+1:

[Trino](https://trino.operate-first.cloud/) is a distributed SQL query engine designed to query large data sets distributed over one or more heterogeneous data sources. Since our data is stored in Ceph, the Operate First Trino is currently configured to use the `opf-datacatalog-bucket` S3 bucket.

After extracting and analyzing the data in Jupyterhub, we use Trino to store our datasets in the database to further interact with it via Cloudbeaver and Superset.

5. **Cloudbeaver -**

[CloudBeaver](http://cloudbeaver-opf-trino.apps.smaug.na.operate-first.cloud/) is a popular web application that provides a rich UI for working and interacting with SQL databases. The data we exported by using Trino can be accessed in the [CloudBeaver UI](http://cloudbeaver-opf-trino.apps.smaug.na.operate-first.cloud/) and can be queried by SQL queries.

6. **Superset -**

> After creating the tables in Trino, we can start creating our dashboards in Superset. Apache Superset is an open source visualization tool that allows users to create interactive dashboards. [Superset](https://superset.operate-first.cloud/) is very simple to use and requires no coding knowledge.

Reviewer 2: Shall we say we require very basic SQL knowledge instead of saying no coding knowledge? wdyt?

Author: Yeah, sure we can do that.
